### DISCUSSION TITLE
How to resolve git-rev-sync no git repository found error

### DISCUSSION THREAD

**Discussion by sunopar** [vercel/vercel#4039] [General]
When deploying repo with ZEIT Now, I get an error is "Error: [git-rev-sync] no git repository found".
The code is 
```javascript
const git = require('git-rev-sync');

module.exports = {
  generateBuildId: () => {
    const buildId = `${git.branch()}-${git.short()}`
    console.log(`buildId: ${buildId}`)
    return buildId
  },
}
```
Is there a solution to this problem.

**Comment by ties-v** [comment]
This makes sense, as there is nog git in the ZEIT now build environment. 
When you run `now`, the source files are uploaded, but without git.

Why do you want to change the nextjs build ID? This is advanced usage for when you deploy a nextjs app to multiple servers yourself. It is not required for ZEIT now. 

If you really want to change the build ID and are using the github (or gitlab or bitbucket) integration you can make use of some environment variables (see the [docs](https://zeit.co/docs/v2/git-integrations/zeit-now-for-github#Included%20Environment%20Variables), section included environment variables). You can then use `NOW_GITHUB_COMMIT_REF` and/or `NOW_GITHUB_COMMIT_SHA` when building on now.
